File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n=nums.size();
        int i=0,j=0,k=n-1;
        for(int i=0;i<n;i++){
            if(nums[i]==0){
                swap(nums[i],nums[j]);
                j++;
            }
        }
        for(int i=j;i<k;){
            if(nums[i]==2){
                swap(nums[i],nums[k]);
                k--;
            }else{
                i++;
            }
        }
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int i,j=0,n=nums.length;
        for(i=0;i<n;i++){
            if(nums[i]==0){
                int temp=nums[j];
                nums[j]=nums[i];
                nums[i]=temp;
                j++;
            }
        }
        for(int k=j;k<n;k++){
            if(nums[k]==1){
                int temp=nums[j];
                nums[j]=nums[k];
                nums[k]=temp;
                j++;
            }
        }
    }
}